R/model-tree.R

```diff
@@ -95,74 +108,90 @@ model_tree <- function(
     .log_df,
     include_info = c("description","star", "tags"),
     color_by = "run",
+    size_by = NULL,
     add_summary = TRUE,
     digits = 3,
     zoomable = FALSE,
     static = FALSE,
     width = NULL,
     height = NULL,
+    font_size = 10,
     ...
 ){
   UseMethod("model_tree")
 }
 
-#' @rdname model_tree
+
 #' @export
 model_tree.character <- function(
     .log_df,
     include_info = c("description","star", "tags"),
     color_by = "run",
+    size_by = NULL,
     add_summary = TRUE,
     digits = 3,
     zoomable = FALSE,
     static = FALSE,
     width = NULL,
     height = NULL,
+    font_size = 10,
     ...
 ){
   checkmate::assert_directory_exists(.log_df)
   .log_df <- run_log(.log_df, ...)
   model_tree(
     .log_df,
-    include_info = include_info, color_by = color_by,
+    include_info = include_info,
+    color_by = color_by, size_by = size_by,
     add_summary = add_summary, digits = digits,
     zoomable = zoomable, static = static,
-    width = width, height = height
+    width = width, height = height,
+    font_size = font_size
   )
 }
 
-#' @rdname model_tree
 #' @export
 model_tree.bbi_log_df <- function(
     .log_df,
-    include_info = c("description","star", "tags"),
+    include_info = c("description", "star", "tags"),
     color_by = "run",
+    size_by = NULL,
     add_summary = TRUE,
     digits = 3,
     zoomable = FALSE,
     static = FALSE,
     width = NULL,
     height = NULL,
+    font_size = 10,
     ...
 ){
   # Make sure required dependencies are installed
   stop_if_tree_missing_deps(static = static)
 
   # Make tree data
-  tree_data <- make_tree_data(.log_df, include_info, color_by, add_summary)
+  # TODO: decide if we want to append color_by and size_by to tooltip automatically
+  # include_info <- unique(c(include_info, color_by, size_by))
+  tree_data <- make_tree_data(.log_df, include_info, color_by, size_by, add_summary)
 
   # Format coloring
   tree_data <- color_tree_by(tree_data, color_by = color_by)
+  tree_attr <- ifelse(is.null(color_by), "leafCount", color_by)
+
+  # Format sizing
+  tree_data <- size_tree_by(tree_data, size_by = size_by)
```

Comment:
`ifelse -> if/else` because the condition is always a scalar. Same for line 178 above (although I know that one is not new to this PR).

Reply:
My understanding was that `if/else` is mainly necessary when the resulting objects can differ based on the condition. I would have thought `ifelse` was appropriate here. Can you explain a bit more?

Reply:
(Sorry for misplacing my comment; it should have been on line 182)

`if`/`else` is the more basic form.  For a scalar condition, I see no reason to use the higher-level `ifelse`.   

Reasons I see to avoid:
 * `ifelse` suggests "vectorized" (at least to this one reader, but I don't think it's just me), so it requires the reader to analyze more closely to realize it's not
 * using `ifelse` for scalars can lead to unintentionally and silently dropping parts of a non-scalar result (example: https://github.com/metrumresearchgroup/mrgvalidate/commit/c42777c67007d4dc5b03e431045e7266b4138f2a)

Reply:
Conditional logic has been removed in favor of attribute setting for both `size_by` and `color_by` as of [71be083](https://github.com/metrumresearchgroup/bbr/pull/720/commits/71be083f056db14eb61f3919b58609ddfcc353f9) (`size_by` addressed in [e2c0909](https://github.com/metrumresearchgroup/bbr/pull/720/commits/e2c0909c88736176ccee06a70505f66231115e22))